File: Client/src/main/java/com/rspsi/editor/model/OsrsTileFlags.java
package com.rspsi.editor.model;

/** Neutral names for the terrain flags used by the OSRS map codec. */
public final class OsrsTileFlags {
    public static final int BLOCK_MAP_SQUARE = 0x1;
    public static final int BRIDGE = 0x2;
    public static final int REMOVE_ROOFS = 0x4;
    /** RuneLite Constants.TILE_FLAG_VIS_BELOW: tile may be visible from a lower scene level. */
    public static final int VIS_BELOW = 0x8;
    /** @deprecated Use {@link #VIS_BELOW}; retained for source compatibility. */
    @Deprecated
    public static final int MINIMAP_BRIDGE = VIS_BELOW;
    /** Tile is hidden from the normal minimap scene pass when either bit is set. */
    public static final int MINIMAP_HIDDEN = 0x18;

    private OsrsTileFlags() {
    }

    public static boolean hasBridge(int flags) {
        return (flags & BRIDGE) != 0;
    }

    public static boolean removesRoofs(int flags) {
        return (flags & REMOVE_ROOFS) != 0;
    }

    public static boolean visibleBelow(int flags) {
        return (flags & VIS_BELOW) != 0;
    }

    public static boolean isBlocked(int flags) {
        return (flags & BLOCK_MAP_SQUARE) != 0;
    }
}

File: Client/src/main/kotlin/com/rspsi/editor/model/OsrsTileFlags.kt
@file:JvmName("OsrsTileFlags")

package com.rspsi.editor.model

/** Neutral names for terrain flags used by the OSRS map codec and authored world model. */

const val BLOCK_MAP_SQUARE: Int = 0x1
const val BRIDGE: Int = 0x2
const val REMOVE_ROOFS: Int = 0x4

/** RuneLite Constants.TILE_FLAG_VIS_BELOW: tile may be visible from a lower scene level. */
const val VIS_BELOW: Int = 0x8

/**
 * Historical alias retained for source/binary compatibility.
 *
 * New code should use [VIS_BELOW]. The alias remains until compatibility callers are removed.
 */
@Deprecated("Use VIS_BELOW")
@field:java.lang.Deprecated
const val MINIMAP_BRIDGE: Int = VIS_BELOW

/** Tile is hidden from the normal minimap scene pass when either bit is set. */
const val MINIMAP_HIDDEN: Int = 0x18

fun hasBridge(flags: Int): Boolean =
    flags and BRIDGE != 0

fun removesRoofs(flags: Int): Boolean =
    flags and REMOVE_ROOFS != 0

fun visibleBelow(flags: Int): Boolean =
    flags and VIS_BELOW != 0

fun isBlocked(flags: Int): Boolean =
    flags and BLOCK_MAP_SQUARE != 0
